File: python/leetcode/count-complete-tree-nodes.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    count = 0
    def traverse(self, root: TreeNode) -> None:
        if root is not None:
            self.count += 1
            self.traverse(root.left)
            self.traverse(root.right)

    def countNodes(self, root: TreeNode) -> int:
        self.traverse(root)
        return self.count

File: java/leetcode/count-complete-tree-nodes.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    private int traverse(TreeNode root, int count) {
        int newCount = count;
        if (root != null) {
            newCount++;
            newCount = traverse(root.left, newCount);
            newCount = traverse(root.right, newCount);
        }
        return newCount;
    }

    public int countNodes(TreeNode root) {
        return traverse(root, 0);
    }
}
